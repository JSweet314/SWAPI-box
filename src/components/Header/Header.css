.header {
  width: 95%;
  margin: auto;
  border-bottom: 1px solid #AAAAAB;
}

.header__h1 {
  font-size: 3rem;
  text-align: center;
}

.header__favBtn {
  position: absolute;
  right: 2rem;
  top: 2rem;
  padding: 1rem;
  cursor: pointer;
}

.header__favCount {
  margin: 0 0 0 1rem;
  padding: 0.5rem;
  background-color: #4B545F;
  color: #fff;
}