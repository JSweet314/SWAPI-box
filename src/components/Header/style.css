.header {
  width: 100%;
  margin: auto;
  border-bottom: 1px solid #AAAAAB;
  background-color: #4B545F;
  position: relative;
  padding: 1rem;
}

.header__h1 {
  font-size: 5rem;
  color: orange;
  text-align: center;
  margin: 0;
  padding: 2rem 2rem 0 2rem;
}

.header__favBtn {
  background-color: gray;
  position: absolute;
  right: 2rem;
  top: 2rem;
  padding: 1rem;
  cursor: pointer;
  text-decoration: none;
  font-size: 1.5rem;
  color: white;
  border: 1px solid black;
}

.header__favBtn.active {
  background:#18212A;
  color: white;  
}

.header__favCount {
  margin: 0 0 0 1rem;
  padding: 0.5rem;
  background-color: #4B545F;
  color: #fff;
  font-size: 1.5rem;
}

.nav {
  padding: 1rem 0;
  display: flex;
  justify-content: center;
  width: 100%;
}

.nav li {
  display: inline-block;
  font-size: 2.5rem;
  margin: 0 2rem;
}

.nav a {
  text-decoration: none;
  background-color: gray;
  border-radius: 10px 10px;
  padding: 1rem 2rem;
  color: #fff;
}

.nav .active {
  background-color: #18212A;
}