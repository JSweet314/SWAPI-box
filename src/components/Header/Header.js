import React from 'react';
import NavLinks from '../NavLinks/index';
import './Header.css';
import PropTypes from 'prop-types';

const Header = ({ numberOfFavorites }) => {
  return (
    <header className="header">
      <h1 className="header__h1">SWAPI-Box</h1>
      <button className="header__favBtn">
        View Favorites 
        <span className="header__favCount">{numberOfFavorites}</span>
      </button>
      <NavLinks />
    </header>
  );
};

Header.propTypes = {
  numberOfFavorites: PropTypes.number.isRequired
};

export default Header;