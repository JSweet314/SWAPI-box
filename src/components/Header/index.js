import React from 'react';
import {NavLink} from 'react-router-dom';
import PropTypes from 'prop-types';
import './style.css';

const Header = ({numberOfFavorites}) => {
  return (
    <header className="header">
      <h1 className="header__h1">SWAPI-Box</h1>
      <ul className="nav">
        <li>
          <NavLink exact to="/">Home</NavLink>
        </li>
        <li>
          <NavLink to="/category/people">People</NavLink>
        </li>
        <li>
          <NavLink to="/category/planets">Planets</NavLink>
        </li>
        <li>
          <NavLink to="/category/vehicles">Vehicles</NavLink>
        </li>
      </ul>
      <NavLink className="header__favBtn" exact to="/category/favorites">
        View Favorites
        <span className="header__favCount">{numberOfFavorites}</span>
      </NavLink>
    </header>
  );
};

Header.propTypes = {
  numberOfFavorites: PropTypes.number.isRequired
};

export default Header;