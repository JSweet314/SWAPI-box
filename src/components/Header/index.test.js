import React from 'react';
import Header from './index';
import {shallow} from 'enzyme';

describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header numberOfFavorites={0} />);
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});