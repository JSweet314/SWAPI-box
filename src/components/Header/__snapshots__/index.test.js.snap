// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    numberOfFavorites={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="header__h1"
>
          SWAPI-Box
</h1>,
        <ul
          className="nav"
>
          <li>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              exact={true}
                              to="/"
                    >
                              Home
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/category/people"
                    >
                              People
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/category/planets"
                    >
                              Planets
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/category/vehicles"
                    >
                              Vehicles
                    </NavLink>
          </li>
</ul>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="header__favBtn"
          exact={true}
          to="/category/favorites"
>
          View Favorites
          <span
                    className="header__favCount"
          />
</NavLink>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWAPI-Box",
          "className": "header__h1",
        },
        "ref": null,
        "rendered": "SWAPI-Box",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            exact={true}
                            to="/"
              >
                            Home
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/category/people"
              >
                            People
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/category/planets"
              >
                            Planets
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/category/vehicles"
              >
                            Vehicles
              </NavLink>
</li>,
          ],
          "className": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                exact={true}
                to="/"
>
                Home
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Home",
                "exact": true,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/category/people"
>
                People
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "People",
                "to": "/category/people",
              },
              "ref": null,
              "rendered": "People",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/category/planets"
>
                Planets
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Planets",
                "to": "/category/planets",
              },
              "ref": null,
              "rendered": "Planets",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/category/vehicles"
>
                Vehicles
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Vehicles",
                "to": "/category/vehicles",
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": Array [
            "View Favorites",
            <span
              className="header__favCount"
/>,
          ],
          "className": "header__favBtn",
          "exact": true,
          "to": "/category/favorites",
        },
        "ref": null,
        "rendered": Array [
          "View Favorites",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
              "className": "header__favCount",
            },
            "ref": null,
            "rendered": 0,
            "type": "span",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="header__h1"
>
            SWAPI-Box
</h1>,
          <ul
            className="nav"
>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    exact={true}
                                    to="/"
                        >
                                    Home
                        </NavLink>
            </li>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/category/people"
                        >
                                    People
                        </NavLink>
            </li>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/category/planets"
                        >
                                    Planets
                        </NavLink>
            </li>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/category/vehicles"
                        >
                                    Vehicles
                        </NavLink>
            </li>
</ul>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="header__favBtn"
            exact={true}
            to="/category/favorites"
>
            View Favorites
            <span
                        className="header__favCount"
            />
</NavLink>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWAPI-Box",
            "className": "header__h1",
          },
          "ref": null,
          "rendered": "SWAPI-Box",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                exact={true}
                                to="/"
                >
                                Home
                </NavLink>
</li>,
              <li>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/category/people"
                >
                                People
                </NavLink>
</li>,
              <li>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/category/planets"
                >
                                Planets
                </NavLink>
</li>,
              <li>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/category/vehicles"
                >
                                Vehicles
                </NavLink>
</li>,
            ],
            "className": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  exact={true}
                  to="/"
>
                  Home
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Home",
                  "exact": true,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/category/people"
>
                  People
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "People",
                  "to": "/category/people",
                },
                "ref": null,
                "rendered": "People",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/category/planets"
>
                  Planets
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Planets",
                  "to": "/category/planets",
                },
                "ref": null,
                "rendered": "Planets",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/category/vehicles"
>
                  Vehicles
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Vehicles",
                  "to": "/category/vehicles",
                },
                "ref": null,
                "rendered": "Vehicles",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": Array [
              "View Favorites",
              <span
                className="header__favCount"
/>,
            ],
            "className": "header__favBtn",
            "exact": true,
            "to": "/category/favorites",
          },
          "ref": null,
          "rendered": Array [
            "View Favorites",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
                "className": "header__favCount",
              },
              "ref": null,
              "rendered": 0,
              "type": "span",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
